oras/provider.py

```diff
@@ -492,8 +492,10 @@ def _get_location(
         if not session_url:
             return session_url
 
-        # Some registries do not return the full registry hostname
-        prefix = f"{self.prefix}://{container.registry}"
+        # Some registries do not return the full registry hostname.  Check that 
+        # the url starts with a protocol scheme, change tracked with:
```

Comment:
We won't want to replace prefix here - for most cases it will be something meaningful. Rather, I think we want:

```python
prefix = f"{self.prefix}://{container.registry}"

if not session_url.startswith('http'):
    # same
```

Reply:
@vsoch prefix isn't used anywhere else in that method.  So retaining the variable doesn't make sense if we're going to just use the static string

Reply:
Wait that would bread the original fix.  I'll update and push.

Reply:
So the endpoint can return a partial url, meaning one that is just the original self.prefix (http or https) and then the container registry URI. If you just define prefix to be http, it's going to miss the entire registry URL and just return a truncated url.

Reply:
Good catch, should be fixed